src/Compilers/CSharp/Test/Semantic/Semantics/SpanStackSafetyTests.cs

```diff
@@ -1008,19 +1009,231 @@ public static async Task<int> M1()
+                // (9,10): error CS4012: Parameters or locals of type 'Span<int>' cannot be declared in async methods or async lambda expressions.
+                //         (Span<int> s1, Span<int> s2) = new Program(); // 1, 2
+                Diagnostic(ErrorCode.ERR_BadSpecialByRefLocal, "Span<int>").WithArguments("System.Span<int>").WithLocation(9, 10),
+                // (9,24): error CS4012: Parameters or locals of type 'Span<int>' cannot be declared in async methods or async lambda expressions.
+                //         (Span<int> s1, Span<int> s2) = new Program(); // 1, 2
+                Diagnostic(ErrorCode.ERR_BadSpecialByRefLocal, "Span<int>").WithArguments("System.Span<int>").WithLocation(9, 24),
+                // (10,14): error CS4012: Parameters or locals of type 'Span<int>' cannot be declared in async methods or async lambda expressions.
+                //         var (s3, s4) = new Program(); // 3, 4
+                Diagnostic(ErrorCode.ERR_BadSpecialByRefLocal, "s3").WithArguments("System.Span<int>").WithLocation(10, 14),
+                // (10,18): error CS4012: Parameters or locals of type 'Span<int>' cannot be declared in async methods or async lambda expressions.
+                //         var (s3, s4) = new Program(); // 3, 4
+                Diagnostic(ErrorCode.ERR_BadSpecialByRefLocal, "s4").WithArguments("System.Span<int>").WithLocation(10, 18),
+                // (11,10): error CS4012: Parameters or locals of type 'Span<int>' cannot be declared in async methods or async lambda expressions.
+                //         (var s5, var s6) = new Program(); // 5, 6
+                Diagnostic(ErrorCode.ERR_BadSpecialByRefLocal, "var").WithArguments("System.Span<int>").WithLocation(11, 10),
+                // (11,18): error CS4012: Parameters or locals of type 'Span<int>' cannot be declared in async methods or async lambda expressions.
+                //         (var s5, var s6) = new Program(); // 5, 6
+                Diagnostic(ErrorCode.ERR_BadSpecialByRefLocal, "var").WithArguments("System.Span<int>").WithLocation(11, 18)
+                );
+        }
+
+        [Fact, WorkItem(62747, "https://github.com/dotnet/roslyn/issues/62747")]
+        public void AsyncLocals_Using()
+        {
+            var src = @"
+using System.Threading.Tasks;
+
+public class Program
+{
+    public async Task M()
+    {
+        using (default(RS)) { } // 1
+        using (var s1 = default(RS)) { } // 2
+        using (RS s2 = default(RS)) { } // 3
+        using RS s3 = default(RS); // 4
+        using var s4 = default(RS); // 5
+
+        await Task.Yield();
+        return;
+    }
+}
+
+public ref struct RS
+{
+    public void Dispose() { }
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net70);
+            comp.VerifyDiagnostics(
+                // (8,16): error CS9104: Values of type 'RS' cannot be used in async methods or async lambda expressions.
```

Comment:
>Values of type 'RS' cannot be used in async methods or async lambda expressions. [](http://example.com/codeflow?start=41&length=80)

I think that this wording is also confusing, and quite possibly plain incorrect. Values of RS can be used in an async method. For example, the following code compiles today, and I think will continue compiling with changes in this PR, but it is using value of `RS` type in an async method:
[https://sharplab.io/#v2:EYLgtghglgdgNAFxFANgHwAIAYAEGCMArANwCwAUBRgMx4BMe+A7BQN4U6d60YAceANhwBZABQBKDl3bkucvAE5BAOgAiAUxQQAnqPziys+ZwDi6hACUAyhLVQAzgAcA9vfUTDxzhiaeuAXylOIMYhaxwzSxtxHABeAD4cABN1ADMIAFcUBENAynIaHAAnNJx7BCKMgGMEHGs2EMKMABYcVQcXNwkcVhw8/yA===](https://sharplab.io/#v2:EYLgtghglgdgNAFxFANgHwAIAYAEGCMArANwCwAUBRgMx4BMe+A7BQN4U6d60YAceANhwBZABQBKDl3bkucvAE5BAOgAiAUxQQAnqPziys+ZwDi6hACUAyhLVQAzgAcA9vfUTDxzhiaeuAXylOIMYhaxwzSxtxHABeAD4cABN1ADMIAFcUBENAynIaHAAnNJx7BCKMgGMEHGs2EMKMABYcVQcXNwkcVhw8/yA===)
``` 
using System.Threading.Tasks;

public class Program
{
    public async Task M()
    {
        await Task.Delay(1);
        GetRS().Dispose();
        return;
    }
    
    static RS GetRS() => default;
}

public ref struct RS
{
    public void Dispose() { }
}
```

The problem is that we cannot spill these types, not every usage requires spilling though.

So, I think the wording should be more specific to the situation. I would use something like: "A using statement resource of type 'RS' cannot be used in async methods or async lambda expressions."

Reply:
Or, if we want to have a more general wording, something like: "Values of type 'RS' cannot be used in async methods or async lambda expressions **in this context**."